pkg/log/log_context.go

```diff
@@ -0,0 +1,79 @@
+package log
+
+import (
+	"context"
+)
+
+var allowList map[string]bool
+
+func AllowContextFields(fields ...string) {
+	if allowList != nil {
+		panic("the log context fields allowed list can only be set once")
+	}
+
+	allowList = map[string]bool{}
+	for _, v := range fields {
+		allowList[v] = true
+	}
+}
+
+func filterAllowList(info F) F {
+	result := F{}
+	if allowList == nil {
+		panic("AllowContextFields must be set in order to use log.Context")
+	}
+
+	for k, v := range info {
+		_, ok := allowList[k]
+
+		if !ok {
+			continue
+		}
+
+		result[k] = v
+	}
+
+	return result
+}
+
+// nolint:gochecknoglobals
+var infoKey = "54be8dc9-91ac-4f77-b90a-70e1ffd74566" //random guid
```

Comment:
I meant switching from fixed string to private type alias + key as the instance of the alias - so that the context key is not accessible from outside. (also public example: https://github.com/open-telemetry/opentelemetry-go/blob/19294aab4cd8b52afc34b77403e7447109c9b858/trace/context.go#L21)

If I know the guid I can access the context data w/o issues + it is too long.

Reply:
Ahh got it.  This looks good, implementing.